src/main/java/uk/gov/hmcts/reform/ucmc/callback/CallbackHandler.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.ucmc.callback;
+
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public abstract class CallbackHandler {
+
+
+    protected abstract Map<CallbackType, Callback> callbacks();
+
+    public abstract List<CaseEvent> handledEvents();
+
+    public void register(Map<String, CallbackHandler> handlers) {
+        handledEvents().forEach(
+            handledEvent -> handlers.put(handledEvent.getValue(), this));
+    }
+
+    public CallbackResponse handle(CallbackParams callbackParams) {
+        return Optional.ofNullable(callbacks().get(callbackParams.getType()))
```

Comment:
Optional.ofNullable could be static import

Reply:
done